### PR TITLE
Event Page Completed

### LINKED ISSUE #8: Event page Happening Now Section

### LINKED ISSUE #9: Event page Upcoming Section

### PR DESCRIPTION
**Pull request by LayanJay**
issues #9 #8 cleared

### CODE REVIEW

File: components/Button.tsx [hunk 1]
```diff
@@ -1,16 +1,17 @@
 interface props {
   value: string
+  buttonStyles?: string
 }
```
Comment (chamodshehanka) [inline components/Button.tsx]: Interface name should be start with capital letter, change this to `ButtonProps`

File: modules/EventPage/HappeningNow.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// components
+import Post from '../../components/Post'
+
+// icons
+import { HiArrowRight } from 'react-icons/hi'
+
+// types
+import { SectionProps } from '../../pages/event'
```
Comment (chamodshehanka) [inline modules/EventPage/HappeningNow.tsx]: why do you import lines with comments? that doesn't necessary

File: components/Post.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      <NextImage
+        classnames="max-w-xs overflow-hidden rounded-lg mb-5"
+        imgStyles="rounded-lg transform hover:scale-105 transition easa-in"
+        src={picture}
+        alt="dummy pic"
+        width={300}
+        height={200}
+        layout="responsive"
+        placeholder="blur"
+      />
```
Comment (chamodshehanka) [inline components/Post.tsx]: @sithumonline no tail semicolon at the end of each line why is that?

File: modules/EventPage/UpcomingSection.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+// components
+import Post from '../../components/Post'
+
+// icons
+import { HiArrowRight } from 'react-icons/hi'
+
+// types
+import { SectionProps } from '../../pages/event'
```
Comment (chamodshehanka) [inline modules/EventPage/UpcomingSection.tsx]: no need these comments

File: pages/event.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+// modules
+import HappeningNow from '../modules/EventPage/HappeningNow'
+import UpcomingSection from '../modules/EventPage/UpcomingSection'
+
+export interface PostProps {
+  userId: number
+  id: number
+  title: string
+  body: string
+}
```
Comment (chamodshehanka) [inline pages/event.tsx:15]: this is not a prop right, change the name to `IPost` or something

File: pages/event.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+export interface PostsProps {
+  posts: PostProps[]
+}
+
+export interface SectionProps {
+  posts: PostProps[]
+  picture: StaticImageData
+}
```
Comment (chamodshehanka) [inline pages/event.tsx:24]: move `SectionProps` to relevant file,

File: pages/event.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  const postUrl = 'https://jsonplaceholder.typicode.com/posts?_start=0&_limit=8'
+
+  const posts = await fetch(postUrl).then((res) => res.json())
+
+  return {
+    props: {
+      posts,
+    },
+  }
+}
```
Comment (chamodshehanka) [inline pages/event.tsx:56]: again no semicolons

**Review by chamodshehanka — CHANGES_REQUESTED** [review]

**Comment by chamodshehanka** [comment]
@LayanJay don't forget to link issues you've fixed with PRs, so whenever this PR got merged those issues also will be closed automatically 😉

File: components/Post.tsx (hunk 3 above)
Comment (sithumonline) [reply components/Post.tsx]: CI passing ne...? So no need `;` 🤔

Comment (chamodshehanka) [reply components/Post.tsx]: > CI passing ne...? So no need `;` 🤔

this is not about CI, as a best practices we should use semicolon, this is not Go

Comment (chamodshehanka) [reply components/Post.tsx]: JS doesn't always know how to intelligently insert semicolons for you.

**Review by sithumonline — CHANGES_REQUESTED** [review]
Remove comments and gaps between imports, fix naming chamod mentioned and `;` if you prefer.

**Comment by chamodshehanka** [comment]
We got a conflict too 😂

File: components/Post.tsx (hunk 3 above)
Comment (LayanJay) [reply components/Post.tsx]: This semicolon thing is strange

File: modules/EventPage/UpcomingSection.tsx (hunk 4 above)
Comment (LayanJay) [reply modules/EventPage/UpcomingSection.tsx]: resolved

File: pages/event.tsx (hunk 5 above)
Comment (LayanJay) [reply pages/event.tsx:15]: Yep, But this interface uses in the Post component for the props. That's why I used the same interface to define server-rendered posts props and the Post Component props

File: pages/event.tsx (hunk 6 above)
Comment (LayanJay) [reply pages/event.tsx:24]: Is it okay to duplicate this interface in Both event modules? As this interface uses in two different places in Event modules

**Comment by sithumonline** [comment]
@Shehanka @LayanJay what we do for `;` ?? 
https://github.com/vercel/next.js/blob/canary/.prettierrc.json#L3 
next js don't care about  `;`

**Comment by LayanJay** [comment]
> @Shehanka @LayanJay what we do for `;` ??
> https://github.com/vercel/next.js/blob/canary/.prettierrc.json#L3
> next js don't care about `;`

@Shehanka @sithumonline  I haven't seen this kind of a build error before in nextjs... If there is a fix, suggest me I'll make the necessary changes on my side.

**Comment by sithumonline** [comment]
> > @Shehanka @LayanJay what we do for `;` ??
> > https://github.com/vercel/next.js/blob/canary/.prettierrc.json#L3
> > next js don't care about `;`
> 
> @Shehanka @sithumonline I haven't seen this kind of a build error before in nextjs... If there is a fix, suggest me I'll make the necessary changes on my side.

That's not a next js error it's from eslint it's all about eslint rules
https://github.com/sliit-foss/rsvp-frontend/blob/main/.eslintrc#L28

**Comment by LayanJay** [comment]
> > > @Shehanka @LayanJay what we do for `;` ??
> > > https://github.com/vercel/next.js/blob/canary/.prettierrc.json#L3
> > > next js don't care about `;`
> > 
> > 
> > @Shehanka @sithumonline I haven't seen this kind of a build error before in nextjs... If there is a fix, suggest me I'll make the necessary changes on my side.
> 
> That's not a next js error it's from eslint it's all about eslint rules
> https://github.com/sliit-foss/rsvp-frontend/blob/main/.eslintrc#L28

@sithumonline Any suggestions to fix this?

**Comment by LayanJay** [comment]
closing this PR as a better version of event page added as PR

**Pull request closed without merging**